src/core/token/TokenManager.ttslua

```diff
@@ -219,8 +221,14 @@ do
       return
     end
 
+    -- this is used to load the correct state for additional resource tokens (e.g. "Ammo")
+    local callback = nil
+    if stateID ~= nil then
```

Comment:
Check for the "resource" type here as well.  If we ended up with mismatched data (e.g. "clue"/"Charge") it would throw an error

Reply:
Good idea! Updated